pkg/controllers/nodeclass/suite_test.go

```diff
@@ -821,6 +820,47 @@ var _ = Describe("NodeClassController", func() {
 		BeforeEach(func() {
 			profileName = instanceprofile.GetProfileName(ctx, fake.DefaultRegion, nodeClass)
 		})
+		It("should fail to delete the launch template", func() {
```

Comment:
```suggestion
		It("should not delete the NodeClass if launch template deletion fails", func() {
```

This seems like it's more ensuring that we don't delete the NodeClass as long as the launch template deletion fails